### ISSUE TITLE
Define parameter names convention and refactor

### ISSUE THREAD

**Issue by EsteveAguilera** [SchibstedSpain/Barista#158]
There are a lot of methods on the project that have the same parameter but using different names.

e.g, here are 3 different methods with the same parameter `@IdRes` using 3 different names:

`assertFocused(@IdRes resId: Int)`
`assertEndDrawerClosed(@IdRes id: Int)`
`assertRecyclerViewItemCount(@IdRes recyclerViewId: Int, expectedItemCount: Int)` 

I think it will be a great idea to define a naming convention to keep the style while the project grows up

I know there's a lot of refactoring being done on the **barista2** branch, but I've already seen this on that branch

**Comment by rocboronat** [comment]
Remember to do the refactor **after** merging Barista 2

**Comment by rocboronat** [comment]
My vote for saying `viewId` instead of `id`.

**Comment by EsteveAguilera** [comment]
My two cents:

| Resource type/Value  | Parameter name| 
| ------------- |-------------|
| View id | `viewId` |
| String resource| `stringId` |
| Color resource| `color` |
|Drawable|`drawable`|
|String| `text`|
|ResourceMatcher| `resId`|
|`int` Items count| `itemsCount`|
|`int` Position|`position`|
|Expexted xxx| `expectedXxxx`(exptectedPosition)| 
|Matcher<View>|`viewMatcher`|

Feel free to copy/paste this table with your suggestions or comments!

**Comment by Sloy** [comment]
Thanks @EsteveAguilera for the list! 😍  

I agree on most of it. Let me suggest some things:
- **viewId**: I suggest `viewId` for generic views, like clickOn(viewId), and `[type]Id` for explicit view types, like `writeTo(editTextId)` or `openDrawer(drawerId)` or `clickListItem(listId)`.
- **color -> colorRes**: To avoid confusion between a rgb color int and a color resource.
- **resId**: We generally use this for allowing both id and string resources. How about something like `idOrStringRes` to be more explicit? I can't think of any other resource that could be allowed here.
- **viewMatcher**: I suggest `viewMatcher` for generic matchers, and optionally `[type]Matcher` for when the view type is important, like drawerMatcher, recyclerMatcher, etc.
- **Expected xxx**: Not sure about this one. I only see it used for RecylerView assertions as expectedItemCount, and in that case I would just name it itemCount. So I suggest omitting this one.

**Comment by EsteveAguilera** [comment]
Ok @Sloy I've applied your suggestions:

| Resource type/Value  | Parameter name| 
| ------------- |-------------|
| Generic View id | `viewId` |
| Explicit View id | `[type]Id` (writeTo(editTextId), openDrawer(drawerId)) |
| String resource| `stringId` |
| Color resource| `colorRes` |
|Drawable|`drawable`|
|String| `text`|
|ResourceMatcher| `idOrStringRes`|
|`int` Items count| `itemsCount`|
|`int` Position|`position`|
|Generic Matcher<View>|`viewMatcher`|
|Matcher<>|`[type]Matcher` (drawerMatcher , recyclerMatcher )|

The only mimimi I've is about the `idOrStringRes`. Not sure about this one. Maybe someone else has another suggestion...

**Comment by SmasSive** [comment]
I agree with all the naming except (obviously) `idOrStringRes`, I know we should document in some way that here we accept both `idRes` or `stringRes`... But, what about documenting it? In certain way we are exposing implementation details by giving that name, maybe is a good idea to leave it as `idRes` and specify in the documentation of the specific public method that it admits id or string res.

**Comment by EsteveAguilera** [comment]
I'm not a big fan of documenting, but if we can not find a better name, +1 to use `idRes` and specify in the documentation that it accepts id or string resources.

PS: Can we get a `discussion` label here? 😉
